src/Data/List/Relation/Unary/Linked/Properties.agda

```diff
@@ -99,3 +104,13 @@ module _ {R : Rel A ℓ} where
   applyDownFrom⁺₂ : ∀ f n → (∀ i → R (f (suc i)) (f i)) →
                     Linked R (applyDownFrom f n)
   applyDownFrom⁺₂ f n Rf = applyDownFrom⁺₁ f n (λ _ → Rf _)
+
+------------------------------------------------------------------------
+-- applyDownFrom
+
+module _ {P : Pred A p} (P? : Decidable P)
+         {R : Rel A ℓ} (trans : Transitive R)
+         where
+
+  filter⁺ : ∀ {xs} → Linked R xs → Linked R (filter P? xs)
+  filter⁺ = AllPairs⇒Linked ∘ AllPairs.filter⁺ P? ∘ Linked⇒AllPairs trans
```

Comment:
This will be terribly inefficient: going from a linear proof
to a quadratic one only to come back to the linear one afterwards.
Is it really annoying to prove by hand?

Reply:
> Is it really annoying to prove by hand?

Yes it is :cry: The problem is the case where the first element passes the filter, but then you have no idea that what the next element to link it to is. It's definitely possible to prove it via the `Sublist` relation, but I didn't want to get distracted with adding a load of machinery to get it working. 

Unless you have a neat solution, I'll open an issue to fix it?